### ISSUE TITLE
purge is not working as expected?

### ISSUE THREAD

**Issue by frudolph77** [tprasadtp/ubuntu-post-install#45]
Hey,

found your cool script in the net, and played around with it in a VM.
I tried to purge the games from your config, but it is not working.

```
 ~/shared  sudo ./after-effects -c ae.yml -d -A
✈ Initialize & run checks
  ➜ Permission checks
  ✔ Running as root 
  ✦ Using local config file
  ✦ Purge packages is set to true.
  ⚠ No checks are being made on the packages in the list. 
  ⚠ Use carefully!. Purging system packages is stupid idea! 
  ⚠ $bool_purge_pkgs=true  <------ HERE
  ➜ Checking dependencies...
  ✦ Hostname      : osboxes
  ✦ OS            : Ubuntu 20.04 LTS
  ✦ Distro        : Ubuntu
  ✦ Code Name     : bionic
  ✦ Arch          : amd64
  ✦ Version/Number: 6.1.0-RC1/610
  ➜ Check for conflicting apps...
  ✔ No conflicts detected. 
  ➜ Checking connectivity
  ✔ Connected! 
✈ Running environment checks... 
✈ Checking for Ubuntu derivates/Debian 
  ➜ This is a LTS release of Ubuntu
  ✔ Keeping the codename as: bionic. 
  ✔ Applied release specific settings and fixes. 
✈ Check if config is defined 
✈ Checking for updates 
  ✔ Running the latest and greatest! 
  ⚠ $bool_purge_pkgs=true  <------ HERE 
✈ Processing configuration... 
  ✦ Config file - ae.yml
  ➜ Parsing config file 
  ⚠ $bool_purge_pkgs=false  <------ HERE 
✈ Check if autopilot is set/required 
  ✔ Autopilot is ENABLED 
  ⚠ $bool_purge_pkgs=false  <------ HERE 
✖ Purge switch is set to false. Did you pass -d along with the command? 
```
After processing the config the variable `$bool_purge_pkgs` was set back to `false`

At least I've to set
```yaml
config:
  # Enabled Tasks
  tasks:
    purge: true
  flags:
    purge_enabled: true
```

to make it working, but according to your documentation

> Warning
It is necessary to pass command line argument -d or set config.flags.purge_enabled: true in config.

only one of these is needed. With `config.flags.purge_enabled: true` the parameter `-d` has no effect.

Kind regards
Frank

**Comment by tprasadtp** [comment]
It is a bug. By extension `-k` is also affected similar to this. It happens because configs are overriding the flags. FYI It only happens if you run it with flag AND setting in config file. If you already have a fix feel free to submit a PR.

**Comment by tprasadtp** [comment]
Should be fixed in v6.2.0
